### PR TITLE
Handle errors on attach properly

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
Fixes: https://github.com/containers/libpod/issues/6575

**Comment by rhatdan** [comment]
@mheon @baude @giuseppe @vrothberg @QiWang19 @jwhonce @ashley-cui PTAL

**Comment by mheon** [comment]
LGTM

File: pkg/domain/infra/tunnel/containers.go [hunk 1]
```diff
@@ -451,19 +451,19 @@ func (ic *ContainerEngine) ContainerExecDetached(ctx context.Context, nameOrID s
...
 
 func startAndAttach(ic *ContainerEngine, name string, detachKeys *string, input, output, errput *os.File) error { //nolint
-	attachErr := make(chan error)
+	var attachErr error
 	attachReady := make(chan bool)
 	go func() {
-		err := containers.Attach(ic.ClientCxt, name, detachKeys, bindings.PFalse, bindings.PTrue, input, output, errput, attachReady)
-		attachErr <- err
+		attachErr = containers.Attach(ic.ClientCxt, name, detachKeys, bindings.PFalse, bindings.PTrue, input, output, errput, attachReady)
+		attachReady <- true
```
Comment (vrothberg) [inline pkg/domain/infra/tunnel/containers.go:458]: This looks wrong. `attachReady` is passed to `containers.Attach(...)` so I think we're in a race.

Comment (mheon) [reply pkg/domain/infra/tunnel/containers.go:458]: I think we actually don't guarantee that attachReady is set if containers.Attach exits with error. We could make that a guarantee of containers.Attach? Or maybe make attachErr a channel and do a select{} on attachReady, attachErr

Comment (rhatdan) [reply pkg/domain/infra/tunnel/containers.go:458]: I guess we could set attachReady twice.

**Comment by rhatdan** [comment]
@vrothberg @mheon Rewritten using select statement.

**Comment by mheon** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
Changes LGTM, but one of the tests looks to be upset.

**Pull request merged**